src/core/device.c

```diff
@@ -187,21 +176,10 @@ static int device_coldplug(Unit *u) {
                  * unplugged. The downside is that PID1 won't generate plugged->dead transitions
                  * for devices that were unplugged during the initrd transition. */
 
-                found = DEVICE_NOT_FOUND;
-                state = DEVICE_DEAD;
+                d->found = DEVICE_NOT_FOUND;
+                d->state = DEVICE_DEAD;
         }
 
-        d->found = found;
-        device_set_state(d, state);
-        return 0;
-}
-
-static void device_catchup(Unit *u) {
-        Device *d = DEVICE(u);
-
-        assert(d);
-
-        /* Second, let's update the state with the enumerated state */
         device_update_found_one(d, d->enumerated_found, DEVICE_FOUND_MASK);
```

Comment:
This does update `Device.found` and `Device.state`, and may not trigger required transactions.
E.g. on reload, if `d->state == DEVICE_PLUGGED` and `d->found == d->enumerated_found` and they have `DEVICE_FOUND_UDEV` flag, then `device_update_found_one()` does not call device_set_state()...

Reply:
Again I'm not seeing which transition that would be needed in your example ? `plugged->plugged` ?

Reply:
We need to call device_set_state() at least once in device_coldplug() or device_catchup(), even there is no transition like `plugged -> plugged`.

Reply:
Why ?